### PR TITLE
Enable+apply isort via ruff/pre-commit

### PR DESCRIPTION
**Pull request by dbast**
This enables [isort](https://pycqa.github.io/isort/) via ruff/pre-commit and applies it fully automatically (no manual changes), which results in sorting of the imports (no additions or removals).

While this looks like a cosmetic thing, the intention is different:
* Sorting in to sections (stdlib, 3rd party libs, own code..): Ensures that any 3rd party / own code does not overwrite imports from stdlib and the sections also help to see what comes from where.
* Alphabetical ordering within sections: helps to find things within sections, when reading the imports.

This now also runs the black hook after ruff, as reordering imports can result in some work for black.

### CODE REVIEW

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -10,15 +10,15 @@ repos:
     hooks:
     -   id: check-toml
     -   id: check-yaml
--   repo: https://github.com/psf/black
-    rev: 24.8.0
-    hooks:
-    -   id: black
 -   repo: https://github.com/astral-sh/ruff-pre-commit
```
Comment (norhh) [inline .pre-commit-config.yaml:13]: wasn't ruff enabled in the previous commit? Do you know why it did not it apply isort previously?

Comment (dbast) [reply .pre-commit-config.yaml:13]: yes, but the previous PR only enabled the `F` ruleset, see https://github.com/Consensys/mythril/blob/develop/pyproject.toml#L7 ... to find and remove unused code/imports.

ruff has many more rules that can be activated, see https://docs.astral.sh/ruff/rules/ 

This PR enables isort by adding the `I` to the activated rules, see https://github.com/dbast/mythril/blob/isort/pyproject.toml#L7

Comment (norhh) [reply .pre-commit-config.yaml:13]: I mean, I don't see that in this PR. Is the `I` in a different PR?

Comment (norhh) [reply .pre-commit-config.yaml:13]: nvm, found it,
Github compressed some files for me

**Review by norhh — APPROVED** [review]
Thanks for the contribution!

**Pull request merged**